executor/adapter.go

```diff
@@ -94,6 +95,11 @@ func schema2ResultFields(schema *expression.Schema, defaultDB string) (rfs []*as
 // next query.
 // If stmt is not nil and chunk with some rows inside, we simply update last query found rows by the number of row in chunk.
 func (a *recordSet) Next(ctx context.Context, chk *chunk.Chunk) error {
+	if span := opentracing.SpanFromContext(ctx); span != nil && span.Tracer() != nil {
+		span1 := span.Tracer().StartSpan("executor.Next", opentracing.ChildOf(span.Context()))
+		defer span1.Finish()
+	}
+
```

Comment:
How about:

```suggestion
var noop = func() {}

func trace(ctx context.Context, operationName string) func() {
	if span := opentracing.SpanFromContext(ctx); span != nil && span.Tracer() != nil {
		span1 := span.Tracer().StartSpan(operationName, opentracing.ChildOf(span.Context()))
		return span1.Finish
	}
	return noop
}

func (a *recordSet) Next(ctx context.Context, chk *chunk.Chunk) error {
     defer trace(ctx, "executor.Next")
```
does we take care about `defer` performance?